core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -795,6 +795,14 @@ object TestUtils extends Logging {
     }, msg = msg, pause = 0L, waitTimeMs = waitTimeMs)
   }
 
+  def subscribeAndWaitForRecords(topic: String, consumer: KafkaConsumer[Array[Byte], Array[Byte]]): Unit = {
```

Comment:
there's no timeout?

Reply:
There is a default timeout in the underlying `pollRecordsUntilTrue` method but it is not exposed in `subscribeAndWaitForRecords`. I will add it for clarity.